packages/cloud-functions/src/cico/Simplex.ts

```diff
@@ -44,92 +37,41 @@ export interface SimplexPaymentData {
   checkoutHtml: string
 }
 
-interface UserInitData {
-  ipAddress: string
-  timestamp: string
-  userAgent: string
-}
-
-const getUserInitData = async (
-  currentIpAddress: string,
-  deviceId: string,
-  userAgent: string
-): Promise<UserInitData> => {
-  const [data] = await bigQuery.query(`
-    SELECT context_ip, device_info_user_agent, timestamp
-    FROM ${bigQueryProjectId}.${bigQueryDataset}.app_launched
-    WHERE user_address = (
-        SELECT user_address
-        FROM ${bigQueryProjectId}.${bigQueryDataset}.app_launched
-        WHERE device_info_unique_id= "${deviceId}"
-        AND user_address IS NOT NULL
-        ORDER BY timestamp DESC
-        LIMIT 1
-    )
-    ORDER BY timestamp ASC
-    LIMIT 1
-  `)
-
-  if (!data.length) {
-    return {
-      ipAddress: currentIpAddress,
-      timestamp: new Date().toISOString(),
-      userAgent,
-    }
-  }
-
-  const { context_ip, device_info_user_agent, timestamp } = data[0]
-
-  const userInitData = {
-    ipAddress: context_ip,
-    timestamp: timestamp.value,
-    userAgent: device_info_user_agent,
-  }
-
-  return userInitData
-}
-
-const getOrCreateUuid = async (userAddress: string) => {
-  let simplexId = await admin
-    .database()
-    .ref(`registrations/${userAddress}/simplexId`)
-    .once('value')
-    .then((snapshot) => snapshot.val())
-
-  if (simplexId) {
-    return simplexId
-  }
-
-  simplexId = uuidv4()
-  await admin.database().ref(`registrations/${userAddress}`).update({ simplexId })
-
-  return simplexId
-}
-
 const Simplex = {
   fetchQuote: async (
     userAddress: string,
-    ipAddress: string,
-    currencyToBuy: CurrencyCode,
-    fiatCurrency: LocalCurrencyCode,
-    amount: number,
+    ipAddress: string | null,
+    currencyToBuy: DigitalAsset,
+    fiatCurrency: FiatCurrency,
+    amount: number | undefined,
     amountIsFiat: boolean
   ) => {
-    const userUuid = await getOrCreateUuid(userAddress)
-
-    const response = await Simplex.post('/wallet/merchant/v2/quote', {
-      end_user_id: userUuid,
-      digital_currency: currencyToBuy,
-      fiat_currency: fiatCurrency,
-      requested_currency: amountIsFiat ? fiatCurrency : currencyToBuy,
-      requested_amount: amount,
-      wallet_id: 'valorapp',
-      client_ip: ipAddress,
-      payment_methods: ['credit_card'],
-    })
-
-    const simplexQuote: SimplexQuote = await response.json()
-    return simplexQuote
+    try {
```

Comment:
Wrapped this in a `try/catch` block and added some check for `null` and `undefined` params now that the data is coming from the server-side and can't enforce type checking